### PR TITLE
feat: proxy client impl

### PR DESCRIPTION
**Pull request by xilosada**
First iteration for the Proxy Client

### CODE REVIEW

File: crates/context/config/src/client/env/proxy/mutate.rs [hunk 1]
```diff
@@ -1,8 +1,86 @@
...
+pub struct ContextProxyMutateRequest<'a, T> {
     client: CallClient<'a, T>,
+    raw_request: ProxyMutateRequest,
+}
+
+#[derive(Debug)]
+struct Mutate<'a> {
+    pub(crate) signing_key: [u8; 32],
+    pub(crate) raw_request: ProxyMutateRequest,
+    _marker: PhantomData<&'a ()>,
```
Comment (xilosada) [inline crates/context/config/src/client/env/proxy/mutate.rs]: @miraclx I'll need some guidance here. Not sure what is the best way to handle the lifetime

Comment (miraclx) [reply crates/context/config/src/client/env/proxy/mutate.rs]: you don't need one here

File: crates/context/config/src/client/env/proxy/mutate.rs [hunk 2]
```diff
@@ -1,8 +1,86 @@
...
+}
+
+impl<'a> Method<Near> for Mutate<'a> {
```
Comment (petarjuki7) [inline crates/context/config/src/client/env/proxy/mutate.rs]: I don't think you need lifetimes here.

File: crates/context/config/src/client/env/proxy/mutate.rs [hunk 3]
```diff
@@ -1,8 +1,86 @@
...
+
+        Ok(encoded)
+    }
+
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        serde_json::from_slice(&response).map_err(Into::into)
+    }
+}
+
+impl<'a> Method<Starknet> for Mutate<'a> {
```
Comment (petarjuki7) [inline crates/context/config/src/client/env/proxy/mutate.rs]: And here

File: crates/context/config/src/client/env/proxy/mutate.rs [hunk 4]
```diff
@@ -1,8 +1,86 @@
...
+}
+
+impl<'a, T: Transport> ContextProxyMutateRequest<'a, T> {
+    pub async fn send(
+        self,
+        signing_key: [u8; 32],
+    ) -> Result<Option<ProposalWithApprovals>, ClientError<T>> {
+        let request = Mutate {
+            signing_key,
+            _marker: PhantomData,
```
Comment (petarjuki7) [inline crates/context/config/src/client/env/proxy/mutate.rs]: No need for a a `_marker`

**Review by petarjuki7 — DISMISSED** [review]

File: crates/context/config/src/client/env/proxy/mutate.rs (hunk 1 above)
Comment (xilosada) [reply crates/context/config/src/client/env/proxy/mutate.rs]: Removed

File: crates/context/config/src/client/env/proxy.rs [hunk 5]
```diff
@@ -7,17 +7,17 @@ use mutate::ContextProxyMutate;
 use query::ContextProxyQuery;
 
 #[derive(Copy, Clone, Debug)]
-pub enum ContextProxy {}
+pub enum ContextConfig {}
```
Comment (miraclx) [inline crates/context/config/src/client/env/proxy.rs:10]: this is the wrong change. it should be `ContextProxy`

**Pull request merged**